pytorch_lightning/core/optimizer.py

```diff
@@ -427,3 +430,18 @@ def zero_grad(self, set_to_none: Optional[bool] = False) -> None:
 
     def __repr__(self) -> str:
         return "No Optimizer"
+
+
+@runtime_checkable
+class _SupportsLRScheduler(Protocol):
+    """This class is used to detect if a learning rate scheduler is supported for default configuration using
+    `isinstance(obj, _SupportsLRScheduler)`."""
+
+    def step(self, *args: Any, **kwargs: Any) -> None:
```

Comment:
Can we make this one more strict? It should raise an expectation with timm schedulers.

Reply:
runtime_checkable doesn't check for types, just check the presence of the methods.
ref: https://docs.python.org/3/library/typing.html#typing.runtime_checkable

I think we should rather check whether the scheduler is an instance of native PT schedulers or not. if it's not, then the user should always override the new hook.